supertokens_python/recipe/session/access_token.py

```diff
@@ -45,45 +45,29 @@ def sanitize_number(n: Any) -> Union[Union[int, float], None]:
 
 def get_info_from_access_token(
     jwt_info: ParsedJWTInfo,
-    jwk_clients: List[JWKClient],
+    combined_jwks: List[PyJWK],
     do_anti_csrf_check: bool,
 ):
     try:
         payload: Optional[Dict[str, Any]] = None
-        client: Optional[JWKClient] = None
-        keys: Optional[List[PyJWK]] = None
-
-        # Get the keys from the first available client
-        for c in jwk_clients:
-            try:
-                client = c
-                keys = c.get_latest_keys()
-                break
-            except JWKSRequestError:
-                continue
-
-        if keys is None or client is None:
-            raise PyJWKClientError("No key found")
-
-        if jwt_info.version < 3:
+        if jwt_info.version >= 3:
+            matching_key = [k for k in combined_jwks if k.key_id == jwt_info.kid][0]  # type: ignore
```

Comment:
Whats the type ignore for?

Reply:
And the rest as well?

Reply:
> type ignore for?

pyjwt lib isn't properly typed so pylint throws errors. `type: ignore` suppresses that error.

Reply:
What happens if no matching key is found?

Reply:
```
matching_key = jwk_client.get_matching_key_from_jwt(
                jwt_info.raw_token_string
            )
```

Actually, now we are using this for v3 tokens. 

```python
try:
    return self.cached_jwks[kid]  # type: ignore
except KeyError:
    if not self.is_cooling_down():
        # One more attempt to fetch the latest keys
        # and then try to find the key again.
        self.fetch()
        try:
            return self.cached_jwks[kid]  # type: ignore
        except KeyError:
            pass
except Exception:
    raise JWKSKeyNotFoundError("No key found for the given kid")

raise JWKSKeyNotFoundError("No key found for the given kid")
```

If it doesn't find a matching key for the kid, it tries 1 more time to refresh (provided cooldown period isn't active right now). If still not found, it throws error